@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -553,6 +555,76 @@ function placePointLabelAtCurrentAnchor(
     textCanvas: TextCanvas,
     screenCollisions: ScreenCollisions,
     isRejected: boolean,
+    outScreenPosition: THREE.Vector3
+): PlacementResult {
+    assert(labelState.element.layoutStyle !== undefined);
+
+    const label = labelState.element;
+    const persistent = labelState.visible;
+    const labelThemeLayout = label.layoutStyle!;
+    const labelStateLayout = labelState.textLayoutState;
+    // Either not persistent or textLayoutState should be already initialized.
+    assert(!persistent || labelStateLayout !== undefined);
+    // Use recently rendered (state stored) layout if available, otherwise theme based style.
+    const placement =
+        persistent && labelStateLayout !== undefined
+            ? labelStateLayout.getPlacement(labelThemeLayout)
+            : { h: labelThemeLayout.horizontalAlignment, v: labelThemeLayout.verticalAlignment };
+    const result = placePointLabelAtAnchor(
+        labelState,
+        screenPosition,
+        placement.h,
+        placement.v,
+        scale,
+        textCanvas,
+        screenCollisions,
+        isRejected,
+        false,
+        outScreenPosition
+    );
+    const basePlacementTest =
+        placement.h === labelThemeLayout.horizontalAlignment &&
+        placement.v === labelThemeLayout.verticalAlignment;
+
+    if (!basePlacementTest) {
+        // Text elements with different then base placement require their own unique layout style.
+        // Provide private copy of label layout and attach it to current label's text canvas.
+        textCanvas.textLayoutStyle = label.layoutStyle!.clone();
+        // Setup TextCanvas layout settings of the new placement as it is required for further
+        // TextBufferObject allocations and measurements in addText().
+        textCanvas.textLayoutStyle.horizontalAlignment = placement.h;
+        textCanvas.textLayoutStyle.verticalAlignment = placement.v;
+    }
```

Comment:
Copypasta from placePointLabelChoosingAnchor(). Please read my comments there. This shouldn't be needed here.